sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/implementation/serializer/DigitalTwinsStringSerializer.java

```diff
@@ -33,8 +34,8 @@ public void serialize(String s, JsonGenerator jsonGenerator, SerializerProvider
 
     private boolean isValidJson(String jsonInString ) {
         try {
-            mapper.readTree(jsonInString);
-            return true;
+            JsonNode node = mapper.readTree(jsonInString);
+            return node.isContainerNode() || node.isTextual();
```

Comment:
q - is this to ensure primitive values are not treated as json (numeric, bool etc)? Can jsonInString be an array?

Reply:
That is correct, Arrays are fine, since only strings will end up here, boolean tokens will also fall into the same category, if someone decides to use "true" or "false" as string values, the previous code would treat them as actual boolean tokens as they are valid json tokens. by making sure that only text or container nodes go through, we will maintain the functionality where we accept the full json payload and do not write json values when they are non-string primitive types.